### PR TITLE
Update ChannelsLiveServerTestCase to fork Daphne as process

### LINKED ISSUE #897: ChannelsLiveServerTestCase ReactorNotRestartable exception
Yes, I updated all channels related stuff this morning :-)

```
Traceback (most recent call last):
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/nose/suite.py", line 210, in run
    self.setUp()
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/nose/suite.py", line 293, in setUp
    self.setupContext(ancestor)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/nose/suite.py", line 316, in setupContext
    try_run(context, names)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/test/testcases.py", line 1352, in setUpClass
    raise cls.server_thread.error
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/channels/testing/live.py", line 28, in run
    self.daphne.run()
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/daphne/server.py", line 132, in run
    reactor.run(installSignalHandlers=self.signal_handlers)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 266, in run
    self.startRunning(installSignalHandlers=installSignalHandlers)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/twisted/internet/base.py", line 1222, in startRunning
    ReactorBase.startRunning(self)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/twisted/internet/base.py", line 730, in startRunning
    raise error.ReactorNotRestartable()
twisted.internet.error.ReactorNotRestartable: 
```

### PR DESCRIPTION
**Pull request by brianmay**
Fixes #897.

### CODE REVIEW

**Comment by andrewgodwin** [comment]
Give it time to load, it sometimes takes a few seconds. Looks like it's complaining about you using single quotes not double quotes (I have it on a very strict set of linters)

**Comment by brianmay** [comment]
Ok, got it now. Initially, Travis was giving me an error when attempting to access the log. Hopefully, Travis will be happy now.

File: channels/testing/live.py [hunk 1]
```diff
@@ -1,91 +1,65 @@
...
-        self.join()
+        return "http://%s:%s" % (self.host, self._port_storage.value)
 
+    @property
+    def live_server_ws_url(self):
 
-class ChannelsLiveServerTestCase(LiveServerTestCase):
-    """
-    Drop-in replacement for Django's LiveServerTestCase.
+        return "ws://%s:%s" % (self.host, self._port_storage.value)
```
Comment (andrewgodwin) [inline channels/testing/live.py]: I'd rather you assigned the value directly to `self.port` after `ready.wait()`

File: channels/testing/live.py [hunk 2]
```diff
@@ -1,91 +1,65 @@
...
+
+        self._server_process.terminate()
+        self._server_process.join()
+        super(ChannelsLiveServerTestCase, self)._post_teardown()
+
+    def _is_in_memory_db(self, connection):
+        """Check if DatabaseWrapper holds in memory database."""
+
+        if connection.vendor == "sqlite":
+            if django.VERSION >= (1, 11):
```
Comment (andrewgodwin) [inline channels/testing/live.py]: No need for the check, Channels is only for Django 1.11 and up.

**Comment by brianmay** [comment]
Addressed feedback.

**Comment by andrewgodwin** [comment]
One last code style nit - can you remove the extra newlines after function definitions? I don't mind having them in function bodies, but it's a bit odd having def, blank line, return compared to the rest of the codebase?

**Comment by brianmay** [comment]
Done. What about the blank line in `_is_in_memory_db()`? Keep or remove it?

**Comment by andrewgodwin** [comment]
Yes, I'd love if that was gone as well, and the docstring was updated to have the text on a newline (all the docstrings in the source have text on a different line from the `"""`, even if it fits on one line)

**Comment by brianmay** [comment]
Done.

**Comment by andrewgodwin** [comment]
Great, thanks!

**Pull request merged**